plugins/importexport/doaj/filter/DOAJJsonFilter.inc.php

```diff
@@ -0,0 +1,178 @@
+			$doi = $pubObject->getStoredPubId('doi');
+			if (!empty($doi)) $article['bibjson']['identifier'][] = array('type' => 'doi', 'id' => $doi);
+			// Print and online ISSN
+			if (!empty($pissn)) $article['bibjson']['identifier'][] = array('type' => 'pissn', 'id' => $pissn);
+			if (!empty($eissn)) $article['bibjson']['identifier'][] = array('type' => 'eissn', 'id' => $eissn);
+			// Year and month from article's publication date
+			$publicationDate = $this->formatDate($issue->getDatePublished());
+			if ($pubObject->getDatePublished()) {
+				$publicationDate = $this->formatDate($pubObject->getDatePublished());
+			}
+			$yearMonth = explode('-', $publicationDate);
+			$article['bibjson']['year'] = $yearMonth[0];
+			$article['bibjson']['month'] = $yearMonth[1];
+			/** --- FirstPage / LastPage (from PubMed plugin)---
+			 * there is some ambiguity for online journals as to what
+			 * "page numbers" are; for example, some journals (eg. JMIR)
+			 * use the "e-location ID" as the "page numbers" in PubMed
+			 */
+			$pages = $pubObject->getPages();
+			if (preg_match("/([0-9]+)\s*-\s*([0-9]+)/i", $pages, $matches)) {
+				// simple pagination (eg. "pp. 3-8")
+				$article['bibjson']['start_page'] = $matches[1];
+				$article['bibjson']['end_page'] = $matches[2];
+			} elseif (preg_match("/(e[0-9]+)/i", $pages, $matches)) {
+				// elocation-id (eg. "e12")
+				$article['bibjson']['start_page'] = $matches[1];
+				$article['bibjson']['end_page'] = $matches[1];
+			}
+			// FullText URL
+			$article['bibjson']['link'] = array();
+			$article['bibjson']['link'][] = array(
+				'url' => Request::url(null, 'article', 'view', $pubObject->getId()),
+				'type' => 'fulltext',
+				'content_type' => 'html'
+			);
+			// Authors: name, email and affiliation
+			$article['bibjson']['author'] = array();
+			$articleAauthors = $pubObject->getAuthors();
+			foreach ($articleAauthors as $articleAauthor) {
+				$author = array('name' => $articleAauthor->getFullName());
+				$email = $articleAauthor->getEmail();
+				if (!empty($email)) $author['email'] = $email;
+				$affiliation = $articleAauthor->getAffiliation($pubObject->getLocale());
+				if (!empty($affiliation)) $author['affiliation'] = $affiliation;
+				$article['bibjson']['author'][] = $author;
+			}
+			// Abstract
+			$abstract = $pubObject->getAbstract($pubObject->getLocale());
+			if (!empty($abstract)) $article['bibjson']['abstract'] = PKPString::html2text($abstract);
+			// Keywords
+			$keywords = array_map('trim', explode(';', $pubObject->getSubject($pubObject->getLocale())));
+			if (!empty($keywords)) $article['bibjson']['keywords'] = $keywords;
+
+			/* not needed here:
+			// Language
+			$language = AppLocale::get3LetterIsoFromLocale($pubObject->getLocale());
+			// publisherRecordId
+			$publisherRecordId = $pubObject->getId();
+			// documentType
+			$type = $pubObject->getType($pubObject->getLocale());
+			*/
+
+			$articles[] = $article;;
+
+		}
+
+		$json = json_encode($articles);
```

Comment:
Argh, yeah, the filter framework is one of the places we're still using references. It could use a review and fix for that, but that's out of scope here.